### PR TITLE
Add "getting started" docs

### LINKED ISSUE #305: add "getting started" docs
**Is your feature request related to a problem? Please describe.**
n/a

**Describe the solution you'd like**
lets add "getting started" docs and link them to the main readme. this should include:
- Generate the protos in your language
- implement a write handler and register it
- generate the COS client, process command, and get state
- implement a read handler and register it

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by zenyui**
resolves #305

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+is created and updated.
+
+### Methods
+`HandleCommand` accepts a command (think "request") and a prior state and returns an event.
+For example, given a command to `UpdateUserEmail` and a User, this RPC might return
+an event called `UserEmailUpdated`. This method is allowed to throw exceptions
+(as gRPC error statuses) if the inbound command (request) is invalid.
+
+`HandleEvent` accepts an event and the prior state of an entity and returns a new state.
+For example, given a UserEmailUpdated event and a User, this RPC would return a
```
Comment (ido-namely) [inline docs/getting-started.md:55]:
```suggestion
For example, given a `UserEmailUpdated` event and a User, this RPC would return a
```

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+new User instance with the email updated. It's advisable to do all your
+validations in `HandleCommand` such that this method is unlikely to encounter
+invalid data. If `HandleCommand` fails, `HandleEvent` is not called for that
+command.
+
+### Registration
```
Comment (ido-namely) [inline docs/getting-started.md:61]: maybe we have this explained somewhere else but I would consider reiterate what is explained in the diagram and then how cos requires some mandatory vars with the write handler vars among them.

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+```yaml
+# set the host and port for your write handler gRPC server
+COS_WRITE_SIDE_HOST: localhost
+COS_WRITE_SIDE_PORT: 50051
+```
+
+(If you are new to setting up COS, see the full [configuration](docs/configuration.md) guide.)
+
+### Sending requests to COS
+Once you have implemneted a write handler and registered it with COS, you are
```
Comment (ido-namely) [inline docs/getting-started.md:73]:
```suggestion
Once you have implemented a write handler and registered it with COS, you are
```

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+[namely/docker-protoc](https://github.com/namely/docker-protoc) with support
+for many languages.
+
+## Implement a "Write Side"
+
+Drawing from CQRS terminology, COS defines a [writeside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/writeside.proto) that you must implement to define how state
+is created and updated.
+
+### Methods
+`HandleCommand` accepts a command (think "request") and a prior state and returns an event.
```
Comment (ido-namely) [inline docs/getting-started.md:49]: nit: 
```suggestion
`HandleCommand` accepts a command (or a "request" if you will) and a prior state and returns an event.
```

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+For example, given a command to `UpdateUserEmail` and a User, this RPC might return
+an event called `UserEmailUpdated`. This method is allowed to throw exceptions
+(as gRPC error statuses) if the inbound command (request) is invalid.
```
Comment (ido-namely) [inline docs/getting-started.md:52]:
```suggestion
an event called `UserEmailUpdated`. This method is encouraged to throw exceptions
(as gRPC error statuses) **if** the inbound command (request) is invalid.
```

File: docs/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+(If you are new to setting up COS, see the full [configuration](docs/configuration.md) guide.)
+
+### Sending requests to COS
+Once you have implemneted a write handler and registered it with COS, you are
+able to start sending requests to chief-of-state with the chief-of-state
+[gRPC client methods](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/service.proto).
+
+`ProcessCommand` accepts a command (request) and an entity ID, and processes
```
Comment (ido-namely) [inline docs/getting-started.md:77]: I think we should reiterate what the entity ID is.
I like the explanation for GetState to think of this as keys and values and I think it's also useful for the context of creating and updating an entity.

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+COS defines a [readside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/readside.proto)
+and allows you to implement many "readsides" (from CQRS "read side") to
+stream your events and state to external destinations. For example, you might
+implement a readside that writes to your applications private elasticsearch
+cluster for serving fancy search queries, and you could implement a second
+readside that publishes your event to kafka for consumption by other services.
+
+The COS readside processor guarantees that events are served in the order they were
+persisted to the journal per entity. Offsets are tracked for per readside, so
```
Comment (ido-namely) [inline docs/getting-started.md:95]: I think we should link or add a few words explaining what a journal is, as part of  the cqrs world

File: docs/getting-started.md (hunk 5 above)
Comment (zenyui) [reply docs/getting-started.md]: don't say i never did nothin for ya

File: docs/getting-started.md (hunk 4 above)
Comment (zenyui) [reply docs/getting-started.md]: don't say i never did nothin for ya

File: docs/getting-started.md (hunk 7 above)
Comment (zenyui) [reply docs/getting-started.md]: Yeah, I wrestled with how in depth this doc should be.

File: docs/getting-started.md (hunk 2 above)
Comment (zenyui) [reply docs/getting-started.md:44]: A few lines down there's a link to the full config doc.

Regarding the diagram, yeah maybe that and a few sentences describing the components at the top. I'm torn about the scope of this doc.

File: docs/getting-started.md (hunk 7 above)
Comment (Tochemey) [reply docs/getting-started.md]: I don't think we should explain what Journal is in the CQRS world. There are a lot of doc out there explaining. We can just elaborate that COS help build event sourced/cqrs driven application and maybe and I mean maybe point to the some docs.

**Review by ido-namely — APPROVED** [review]

**Pull request merged**